### DISCUSSION TITLE
能否增加URL类型文档和属性，方便制作成收藏夹。

### DISCUSSION THREAD

**Discussion by LeXwDeX** [toeverything/AFFiNE#6669] [Ideas]
### Description
能否增加URL类型的文档，以及文档中能否增加URL类型的属性，这样可以把URL作为收藏夹使用，并且也可单独检索具有URL属性的成员。

### Use case

希望AFFiNE在未来可以成为替代浏览器收藏夹成为比较重要的产品，首先是因为URL是一个重要的资源，它对于知识库概念来说具有多个属性：可以成为单纯的URL作为浏览器收藏夹功能的成员，也可以成为文档的元数据（类似于论文的引用）成员，也可以作为关联文档的索引（使用Domain或完整URL字符进行关联文档查询）。

还有就是，浏览器收藏夹的简易树形结构无法满足上百个URL文件的管理了，主要是因为（与上面提到的情况一致）：
* 有些URL之间是具有相关性的。
* 有些URL不能用简单的树形结构归类。
* 有些收藏夹的URL和知识库的文档之间也是具有相关性的。

所以希望把URL可以单独的作为一个资源管理，使其可以被Tag，可以被增加其他属性（比如：简述），标题。
当然也具有原始的文件夹树状关系，让用户导出收藏夹数据到AFFiNE系统内，最后用户可以把URL被其他文档引用，也可以单纯的作为收藏夹使用。

之后，如果上述功能拓展下来，那么还可以开发一个带简易展示和搜索功能的浏览器插件，去替代掉浏览器的原始收藏夹功能：
* 可用常规的方式检索：索引名称，索引DOMAIN和URL元数据。
* 可树状结构展开，方便当朴素的收藏夹功能使用。

**Comment by HeJiachen-PM** [comment]
如果我们将url实现为一种blocks（也即扩展当前的bookmark），而后允许database与collection管理这些url，是不是也能满足你的建议？

**Reply by LeXwDeX** [reply]
是的，其实只要是能单独管理和关联就行了。
